flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificInputTypeStrategies.java

```diff
@@ -76,6 +77,18 @@
                                             logical(LogicalTypeRoot.BOOLEAN),
                                             logical(LogicalTypeFamily.NUMERIC))));
 
+    /** Input strategy for {@link BuiltInFunctionDefinitions#JSON_ARRAY}. */
+    public static final InputTypeStrategy JSON_ARRAY =
+            varyingSequence(
+                    symbol(JsonOnNull.class),
+                    or(
+                            logical(LogicalTypeFamily.CHARACTER_STRING),
+                            logical(LogicalTypeFamily.BINARY_STRING),
+                            logical(LogicalTypeFamily.TIMESTAMP),
+                            logical(LogicalTypeFamily.CONSTRUCTED),
+                            logical(LogicalTypeRoot.BOOLEAN),
+                            logical(LogicalTypeFamily.NUMERIC)));
```

Comment:
Can the arguments include also `JSON_OBJECTs`?

Reply:
Yes, and since `JSON_OBJECT` returns a character string, this is covered here.

Reply:
I'd suggest then to add an example using this in the relevant places, what do you think?

Reply:
Can you elaborate what you mean? I've already updated the docs to state that this is possible including an example.

Reply:
I mean in the places where you list some examples, like here: https://github.com/apache/flink/pull/17313/files#diff-539fb22ee6aeee4cf07230bb4155500c6680c4cc889260e2c58bfa9d63fb7de5R775
to also include an example of an array with nested json objects.